src/Microsoft.DotNet.Darc/DarcLib/Helpers/DependencyFileManager.cs

```diff
@@ -39,8 +40,10 @@ public class DependencyFileManager : IDependencyFileManager
         { "dotnet", "tools" },
     }.ToImmutableDictionary();
 
-    private readonly IGitRepo _gitClient;
+    private IGitRepo GetGitClient(string repoUri) => _gitClientFactory(repoUri);
+
     private readonly IVersionDetailsParser _versionDetailsParser;
+    private readonly Func<string, IGitRepo> _gitClientFactory;
```

Comment:
Wouldn't it be cleaner to define the full type for the factory?

Reply:
We'd need to then create some arbitrary implementation to make it work with the old code